### PR TITLE
Made ChemMaster Sorted

### PR DESCRIPTION
**Pull request by DmitriTheDemon**
## About the PR
The ChemMaster reagent list is now sorted

## Why / Balance
Hello there, gentle medics of Frontier. Does your ChemMaster look like a toddler's bedroom after a sugar binge? Are you tired through a huge list of chemicals you have saved looking for the one you need? Well look no further with the new ChemMaster Deluxe Sortable Reagents!

Now your reagents will all fall right into line for your browsing convenience, making chemistry on the go faster and easier than ever before.

## How to test
Obtain ChemMaster
Insert chemicals in any order
Observe reagent list is sorted

## Media
Chemmaster Sorted

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

N/A

**Changelog**

:cl:
- tweak: ChemMaster reagent list is now sorted alphabetically!

### CODE REVIEW

**Comment by dvir001** [comment]
This should be upstream PR, or at minimum more comments and less spaces remove / added to random lines

**Comment by DmitriTheDemon** [comment]
> This should be upstream PR, or at minimum more comments and less spaces remove / added to random lines

The style guide rules put into the code called those out as being misformatted, so I was trying to clean up a little tech debt while I was in there but I can omit it if you prefer.

**Comment by whatston3** [comment]
> The style guide rules put into the code called those out as being misformatted, so I was trying to clean up a little tech debt while I was in there but I can omit it if you prefer.

As-is, not really "our" tech debt.

Minimally, these changes should be a oneliner, no?
ChemMasterWindow.xaml.cs:185:
`foreach (var (reagent, quantity) in state.BufferReagents.OrderBy(x => x.Reagent.Prototype))`

Even with this minimal set of changes, seems to work fine:
image

**Comment by DmitriTheDemon** [comment]
> .OrderBy(x => x.Reagent.Prototype)

Changed to use this

File: Content.Client/Chemistry/UI/ChemMasterWindow.xaml.cs [hunk 1]
```diff
@@ -182,7 +182,7 @@ private void UpdatePanelInfo(ChemMasterBoundUserInterfaceState state)
             };
             bufferHBox.AddChild(bufferVol);
 
-            foreach (var (reagent, quantity) in state.BufferReagents)
+            foreach (var (reagent, quantity) in state.BufferReagents.OrderBy(x => x.Reagent.Prototype))
```
Comment (whatston3) [inline Content.Client/Chemistry/UI/ChemMasterWindow.xaml.cs:185]:
```suggestion
            foreach (var (reagent, quantity) in state.BufferReagents.OrderBy(x => x.Reagent.Prototype)) // Frontier: add OrderBy
```

**Review by whatston3 — APPROVED** [review]
Looks good to me.

**Comment by muburu** [comment]
it'd be nice if they were color coded as well like they are in the dispenser

**Comment by dvir001** [comment]
Single line change me happy.

File format is upstream issue.

**Comment by DmitriTheDemon** [comment]
> it'd be nice if they were color coded as well like they are in the dispenser

I'll look into that

**Pull request merged**